application-templates/starter/src/components/channel-details/transform-errors.js

```diff
@@ -0,0 +1,39 @@
+import omitEmpty from 'omit-empty-es';
+
+const withoutEmptyErrorsByField = (errorsByField) => omitEmpty(errorsByField);
+
+const DUPLICATE_FIELD_ERROR_CODE = 'DuplicateField';
+
+/**
+ * MappedErrors { [errorCodeKey]: any }
+ * TransformedErrors: { mappedErrors: MappedErrors, unmappedErrors: [], formError: [] }
+ *
+ * @param  Object graphqlError
+ * @return Object TransformedErrors
+ */
+
+export const transformErrors = (graphQlErrors) => {
+  const errorsToMap = Array.isArray(graphQlErrors)
+    ? graphQlErrors
+    : [graphQlErrors];
+
+  const formErrors = { key: {} }; // will be mappped to form field error messages
+  const unmappedErrors = []; // will result in dispatching `showApiErrorNotification`
+  const mappedErrors = {}; // may contain values mapped from GraphQL errors that would result in dispatching side domain error notification
+
+  errorsToMap.forEach((graphQlError) => {
+    const errorCode = graphQlError?.extensions?.code ?? graphQlError.code;
+    if (errorCode === DUPLICATE_FIELD_ERROR_CODE) {
+      formErrors.key.duplicate = true;
+    } else {
+      unmappedErrors.push(graphQlError);
+    }
+    // as pointed above we might also include checking for certain GraphQL error codes and map them to `mappedErrors` object entries
+  });
```

Comment:
I'm sorry but I think I'm missing something because I don't understand the need for having the `key` prop in the errors object in advance.
I was taking a look at form validation [docs](https://docs.commercetools.com/custom-applications/development/forms#form-validation) and I can't find the reasoning for that.

Reply:
Thanks Carlos, I guess this part (mapping gql errors to form fields validation errors) is not documented in the forms docs so I tried to keep to the conventions I found. I guess you're right so I refactored a little ([9f87612](https://github.com/commercetools/merchant-center-application-kit/pull/2550/commits/9f8761232965e2633df6c876390da3b286227bb9))

Reply:
The main reason that we define the keys in advance is to make it easier to mutate the object when we need to assign error keys to a specific field.